tests/Extension/CoreTest.php

```diff
@@ -306,6 +306,12 @@ public function provideCompareCases()
             [-1, 42, 'abc42'],
             [-1, 0, 'abc42'],
 
+            [0, 42.0, '   42.0'],
+            [0, 42.0, '42.0   '],
+            [-1, 42.0, '42.0abc'],
```

Comment:
PHP 8 returns 0

Reply:
@fabpot: But that's unrelated to this PR. Indeed, https://github.com/php/php-src/pull/3886 hasn't been merged (yet), so several _existing_ tests currently return a different result (https://3v4l.org/dX6Q9/rfc#output).

This PR fixes `{{ 42.0 == '42.0 ' ? 'OK' : 'KO' }}` vs `{{ 42 == '42 ' ? 'OK' : 'KO' }}` (see https://twigfiddle.com/wni568 and compare 3.x with 2.x/1.x)